prov/gni/Makefile.am

```diff
@@ -0,0 +1,123 @@
+# Makefile.am for gni provider
+
+if HAVE_GNI
+
+#
+# want to keep ccan in the include path seen in the
+# source code, so just add prov/gni to the include path
+# rather than prov/gni/ccan
+#
+AM_CFLAGS += -I$(top_srcdir)/prov/gni/include -I$(top_srcdir)/prov/gni/common -I$(top_srcdir)/prov/gni
+
+_gni_files = \
+	prov/gni/common/rbtree.c \
+	prov/gni/common/rbtree.h \
+	prov/gni/common/fasthash/fasthash.c \
+	prov/gni/common/fasthash/fasthash.h \
```

Comment:
Are these generic enough to be placed into common code, so that other providers can use them, if needed?  There are at least 2 other hash implementations, but this would be the first rbtree.